node/src/types/block.rs

```diff
@@ -1457,6 +1451,26 @@ impl Block {
         Block::random_with_deploys(rng, None)
     }
 
+    /// Generates a random switch block.
+    #[cfg(any(feature = "testing", test))]
+    pub fn random_switch_block(rng: &mut TestRng) -> Self {
+        let mut block = Block::random(rng);
+        let next_era_validator_weights = BTreeMap::<PublicKey, U512>::new();
+        block.header.era_end = Some(EraEnd::new(
+            random_era_report(rng),
+            next_era_validator_weights,
+        ));
+        block
+    }
+
+    /// Generates a random non-switch block.
+    #[cfg(any(feature = "testing", test))]
+    pub fn random_non_switch_block(rng: &mut TestRng) -> Self {
+        let mut block = Block::random(rng);
+        block.header.era_end = None;
+        block
+    }
+
```

Comment:
As above, I'd prefer if these were either removed, or were implemented by calling though to `Block::random_with_specifics`, since these generate invalid blocks.

Reply:
Addressed in 517f01d82ec1e1a611834595cfe8253984cd8c26
I decided to keep the helper functions, but they now use the `random_with_specifics()` internally.